darknet_ros/src/YoloObjectDetector.cpp

```diff
@@ -162,7 +162,17 @@ void YoloObjectDetector::cameraCallback(const sensor_msgs::ImageConstPtr& msg) {
   cv_bridge::CvImagePtr cam_image;
 
   try {
-    cam_image = cv_bridge::toCvCopy(msg, msg->encoding);
+    if (msg->encoding == "mono8" || msg->encoding == "bgr8" || msg->encoding == "rgb8") {
+      cam_image = cv_bridge::toCvCopy(msg, msg->encoding);
+    } else if ( msg->encoding == "bgra8") {
+      cam_image = cv_bridge::toCvCopy(msg, "bgr8");
+    } else if ( msg->encoding == "rgba8") {
+      cam_image = cv_bridge::toCvCopy(msg, "rgb8");
+    } else if ( msg->encoding == "mono16") {
+      cam_image = cv_bridge::toCvCopy(msg, "mono8");
```

Comment:
shouldn't this be mono16 ?

Reply:
Well, I couldn't figure out to make darknet work with 16 bit grayscale, so I decided to convert mono16 to mono8. toCvCopy does that.

Reply:
I see! I guess that it will be some byte-wise interleaved copying which might incur a slight performance penalty. What about adding a `ROS_WARN_ONCE` here?

Reply:
I agree, there would a slight performance penalty, it is basically just scaling all pixels by 2^(-8). But leaving a ROS_WARN_ONCE is not a bad idea.